bootstraps/stage4/spack.in.yaml

```diff
@@ -0,0 +1,74 @@
+# This is a Spack Environment file.
+#
+# It describes a set of packages to be installed, along with
+# configuration settings.
+spack:
+  # add package specs to the `specs` list
+  specs:
+  # glibc from stage3, must match the gcc built there or we get two mixed
+  # together, use hash
+  - glibc @GLIBC_HASH@ %gcc@=13.2.0  arch=linux-spack-x86_64
+  - binutils +gold +lto arch=linux-spack-x86_64
+  - gcc +binutils +piclibs ~bootstrap languages=c,c++,fortran,lto arch=linux-spack-x86_64
+  # base system
+  - m4 arch=linux-spack-x86_64
```

Comment:
Drop all hard-coded `arch=linux-spack-=x86_64` in favor of a single requirement `target=x86_64,aarch64 os=spack`

Reply:
Will see if that works, last I tried applying os=spack that way did not work. Might with the override now that I think about it, good thought.

Reply:
The requirement works.  The preferences interface I was using before didn't, this helped clean up a lot of messy complexity.